client/src/App.tsx

```diff
@@ -13,57 +17,65 @@ import SignIn from './pages/SignIn';
 const App: React.FC = () => {
   return (
     <Router>
-      <div className="App">
-        <nav>
-          <ul>
-            <li>
-              <Link to="/">Home</Link>
-            </li>
-            <li>
-              <Link to="/catalog">Equipment Catalog</Link>
-            </li>
-            <li>
-              <Link to="/equipmentbrowser">Equipment Browser</Link>
-            </li>
-            <li>
-              <Link to="/memberdashboard">MemberDashboard</Link>
-            </li>
-            <li>
-              <Link to="/admin/dashboard">Admin Dashboard</Link>
-            </li>
-            <li>
-              <Link to="/admin/equipment-request">Equipment Request</Link>
-            </li>
-            <li>
-              <Link to="/signin">Sign In</Link>
-            </li>
-          </ul>
-        </nav>
+      <StoreProvider>
+        <LoginShield fallback={<SignIn />}>
+          <div className="App">
+            <nav>
+              <ul>
+                <li>
+                  <Link to="/">Home</Link>
+                </li>
+                <li>
+                  <Link to="/catalog">Equipment Catalog</Link>
+                </li>
+                <li>
+                  <Link to="/equipmentbrowser">Equipment Browser</Link>
+                </li>
+                <li>
+                  <Link to="/memberdashboard">MemberDashboard</Link>
+                </li>
+                <li>
+                  <Link to="/admin/dashboard">Admin Dashboard</Link>
+                </li>
+                <li>
+                  <Link to="/admin/equipment-request">Equipment Request</Link>
+                </li>
+              </ul>
+            </nav>
+
+            {React.createElement(function LogOutButton() {
+              const { dispatch } = useStore();
+              const logout = () => dispatch({ type: 'logout' });
+              return (
+                <Button variant="gray" onClick={logout}>
+                  Log out
+                </Button>
+              );
+            })}
```

Comment:
This is basically like a single-use IIFE for defining something that is technically a child component. I meant it as a temporary solution to a "log out" button to use in testing before we have a real log out button in the app. I didn't want to add a whole new component for a temporary log out button to use in testing, but the `useStore()` hook can't be used by the `App` component because the `App` component doesn't have the `StoreProvider` as an ancestor.

Defining this immediately-instantiated function component creates a space that can attach to the global store through the context API, since this `LogOutButton` component does have the `StoreProvider` as an ancestor.

Reply:
Alright, to clarify this is temporary?

Since we are using contexts anyways, I think it would be worthwhile to eventually (soon) have a logout button that uses `Button` internally or a logout HOC that makes anything work as a logout button.